src/main/scala/inox/ast/Paths.scala

```diff
@@ -214,8 +220,9 @@ trait Paths { self: SymbolOps with TypeOps =>
       */
     private def fold[T](base: T, combineLet: (ValDef, Expr, T) => T, combineCond: (Expr, T) => T)
                        (elems: Seq[Element]): T = elems.foldRight(base) {
-      case (Left((id, e)), res) => combineLet(id, e, res)
-      case (Right(e), res) => combineCond(e, res)
+      case (CloseBound(vd, e), res) => combineLet(vd, e, res)
+      case (Condition(e), res) => combineCond(e, res)
+      case (OpenBound(_), res) => res // FIXME should it also take a combiner for OpenBound?
```

Comment:
I don't think `OpenBound` can be combined in any useful way. If we come up with a use-case later on, it'll be easy to extend the `fold` function at that time.